### DISCUSSION TITLE
PlatinText为何在重构后它的伴生对象没有实现MessageKey

### DISCUSSION THREAD

**Discussion by LovesAsuna** [mamoe/mirai#903] [Q&A] [answered]
这将导致我messagechain.get无法迁移

**Comment by Him188** [comment, marked as answer]
由于一条消息不一定只包含一个 `PlainText`。`MessageChain.get(PlainText)` 以前是返回第一个实例，这可能会产生迷惑性，保守起见就删除了。

若要获得纯文本内容，可以使用 `message.contentToString()` (Java) 和 `message.content` (Kotlin)

对于其他消息类型，请基于 `MessageChain` 实现的 `List<SingleMessage>` 操作。

**Comment by Him188** [comment]
我也在考虑要不要把这种 get 加回去，需要更多人的意见

**Reply by sandtechnology** [reply]
其实对于get本身我其实建议是返回列表而不是单一实例，因为messagechain来说get本身应该是筛选信息的（
EDIT:不知道当初加入这个方法是出于什么用意

**Comment by LovesAsuna** [comment]
但是其他类型的消息都能用这个get这与你说的迷惑性不是有冲突吗

**Reply by Him188** [reply]
目前只有 ConstrainSingle 可以用，它是单一存在的

**Comment by StageGuard** [comment]
可以用 messageChain.firstIsInstanceOrNull<PlainText>()
